remote/remote_test.go

```diff
@@ -1496,3 +1510,61 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 }
+
+func TestRemoteRetryLogic(t *testing.T) {
```

Comment:
This is awesome! My one nit is if you could please convert this to using `spec` notation and nest it under this `when`: https://github.com/buildpacks/imgutil/blob/77564c120f1445ad6c9c0ca3dfd8de99be016279/remote/remote_test.go#L63

That way it's easier to find the test and debug with consistent notation (like `it.Focus`).

Reply:
Hi @natalieparellano  you mean do not using the table driven approach, right?

Reply:
I think what Natalie means is to keep the convention that we use in our unit tests:

As you can see in line 48 in your code, we have the `TestRemote` function. At the end of this function we call:
`spec.Run(t, "Image", testImage, spec.Sequential(), spec.Report(report.Terminal{}))`.
Then in the `testImage` function, we're testing (most of) the public functions of `remote.go`. Each one under a different `when` section.
Since your changes in `remote.go` are part of the `NewImage` flow, it would be great if you could put your tests under the `NewImage` section (`when("#NewImage", func()` in line 77), while using nested `when` and `it` for each `testCase`.

@natalieparellano, please let me know if I misunderstood what you meant.